### ISSUE TITLE
App rejected because of NSPhotoLibraryUsageDescription 

### ISSUE THREAD

**Issue by NightOwlCoder** [bitstadium/HockeySDK-Xamarin#68]
Reading around, I've found lots of references to "update to SDK 4.1.1" and the Feedback feature is gone.

But I just got this now, and I'm 100% sure we are using 4.1.1.

> Missing Info.plist key - This app attempts to access privacy-sensitive data without a usage description. The app's Info.plist must contain an NSPhotoLibraryUsageDescription key with a string value explaining to the user how the app uses this data. 

Anyone has any idea how this could be happening?

**Comment by Cheesebaron** [comment]
And you are sure you are not using Photo Library in your app?

**Comment by ranterle** [comment]
Hi @NightOwlCoder and @Cheesebaron, to clear some things up: Since iOS 10, you need to include the `NSPhotoLibraryUsageDescription` key in your `Info.plist`, if you use the feedback feature. We dropped it  from the default CocoaPods configuration in the 4.1.1 release of our iOS SDK release. See the [release notes](https://github.com/bitstadium/HockeySDK-iOS/releases/tag/4.1.1) for more information.
However, version 4.1.1 of our Xamarin SDK includes version 4.1.3 of the iOS SDK using **all** features (there is no special build without the feedback module for Xamarin right now), see the [release notes](https://github.com/bitstadium/HockeySDK-Xamarin/releases/tag/4.1.1) for more information.

TL;DR: If you're using the Xamarin SDK and not the crash only version, you need to include the key in your `Info.plist`.

**Comment by NightOwlCoder** [comment]
I find it weird that the best features (crash+telemetry) got tangled with a feedback one. :)

I can't find the crash only nugget package. Any hints for the right name?
Also, I'm hopping "crash only" does include telemetry, right?

**Comment by TroubleMakerBen** [comment]
Hey @NightOwlCoder,

>I can't find the crash only nugget package. Any hints for the right name?

There is no nuget-package with crash-only, there's only a [crash-only component](https://components.xamarin.com/view/hockeyappios-crashonly).

>Also, I'm hopping "crash only" does include telemetry, right?

No, it is *crash only*, if you want to use the telemetry feature, please include the full-featured SDK and include the 'NSPhotoLibraryUsageDescription' even if you are not using the Feedback feature.

Best,
Benjamin

**Comment by NightOwlCoder** [comment]
Wow!

I will never be able to convince my manager to add "we are using your photos" to our app! 😢 

Looks like HockeyApp is out for us..

Thanks for the help.

**Comment by Cheesebaron** [comment]
It is for run-time permission dialogs. It is not like anyone will see them unless you actually use the Feedback functionality.

**Comment by NightOwlCoder** [comment]
Oh, I think I know what you mean.

Even if we add 
```NSPhotoLibraryUsageDescription="please let us access the photo library"```

it would never be seen by our users, as we are not using the feedback feature.

I think I can convince him to leave with that.

Thanks for the help.

**Comment by TroubleMakerBen** [comment]
Yes, what @Cheesebaron said!
It will only show up if you use the feedback-feature and only if a user tries to attach a screenshot from their photo library.

I'm glad we got this resolved. =)
FYI: The only way we could prevent to put this additional work on you would be a completely modularized Hockey SDK. We won't be able to do that for HockeyApp, but we're building [Mobile Center](https://www.hockeyapp.net/blog/2016/11/16/mobile-center-next-generation-hockeyapp.html) as a modular SDK for the exact reason to not have devs add stuff they don't need.

Take care and get in touch if you run into issues!

🖖
Benny
